source/github-helpers/github-token.ts

```diff
@@ -0,0 +1,46 @@
+import {v3} from './api.js';
+import optionsStorage from '../options-storage.js';
+
+const settings = optionsStorage.getAll();
+
+export async function getToken(): Promise<string | undefined> {
+	const {personalToken} = await settings;
+	return personalToken;
+}
+
+export async function expectToken(): Promise<string> {
+	const personalToken = await getToken();
+	if (!personalToken) {
+		throw new Error('Personal token required for this feature');
+	}
+
+	return personalToken;
+}
+
+export async function parseTokenScopes(headers: Headers): Promise<string[]> {
+	// If `X-OAuth-Scopes` is not present, the token may be not a classic token.
+	const scopesHeader = headers.get('X-OAuth-Scopes');
+	if (!scopesHeader) {
+		return [];
```

Comment:
What does this do in practice? Avoids the check and just runs the feature anyway? Or does it break all features that require a specific token?

Reply:
This function just 'parse' scopes.
So, running the feature or requiring a specific token is not Interest of it.

`options.tsx` can pass fetch resp's headers, to parse available scopes, and using parsed result (checking permissions, ...), is its freedom.

`expectTokenScope()` can use this function to spray scopes. (like - if project scope exists, append read:project additionally)
throwing error or return boolean with checking includes is its freedom too.

Reply:
Ok but in the end what happens when a feature expects a scope that this does not return?

Reply:
When we need that, change this function, but not now.
Currently, we are checking is valid token with appending our custom scope 'valid_token', so we can approach same result. But returning null or undefined for future is just making confuses.

Let's make the simple thing it works. Do not consider too further future. IMO, checking if a header exists may not be necessary forever.